src/programs/pkexec.c

```diff
@@ -664,6 +664,28 @@ main (int argc, char *argv[])
         argv[n] = path;
       }
     }
+  if (1)
+    {
+#if _POSIX_C_SOURCE >= 200809L
+      s = realpath(path, NULL);
+#else
+      s = NULL;
+# error We have to deal with realpath(3) PATH_MAX madness
+#endif
+      if (s)
+        {
+          g_free (path);
+          path = g_strdup (s);
+          free (s);
+          /* argc<2 and pkexec runs just shell, argv is guaranteed to be null-terminated.
+           * /-less shell shouldn't happen, but let's be defensive and don't write to null-termination
+           */
+          if (argv[n] != NULL)
```

Comment:
Please, do never ever replace values in argc/argv. I know it's there somewhere above, but that is a remnant of an ancient code and it caused real trouble once. The original argv is not to be touched.

Reply:
That's interesting.

I wouldn't touch it either, but I figured something important was going on with argv[n]:
- seeing that the code that resolves a relative path to an absolute one replaces argv[n]
- then a replacement of a non-real path with a resolved one should also update argv[n]

Should we remove the old remnant instead? Is it only there for backwards compatibility? In which case a comment can be helpful to the next guy. (And maybe a deprecation.)

As for g_free vs free: I didn't look into whether they are actually the same. But I figured this should cover things if they are different. If I look now, I see:
> g_free is like free, but checks for a NULL-pointer.
So.. I guess it should be simply:
```c
          g_free (path);
          path = s;
```
And as for the security implications:
- resolving to the real path at the start of the program and keeping that path later one, should be fine, I reckon.

Reply:
> I figured something important was going on with argv[n]:

The last thing I wanted to do was open up the possibility of something using the original path later on. Because that _would_ be a security thing.

Reply:
@jrybar-rh : updated with comments. Let me know.